iphone/Classes/TiUIApplication.m

```diff
@@ -0,0 +1,16 @@
+#import "TiUIApplication.h"
+
+@implementation TiUIApplication
+
+- (void)sendEvent:(UIEvent *)event
+{
+  [super sendEvent:event];
+
+  for (UITouch *touch in event.allTouches) {
+    if (touch.phase == UITouchPhaseBegan) {
+      [[NSNotificationCenter defaultCenter] postNotificationName:@"kTiUserInteraction" object:nil];
```

Comment:
I am starting to become unsure if we should really have this in the core. I understand the purpose for you, but this was requested by maybe 5 devs over the last years, and it must have some kind of performance hit if we post an internal notification for every tap that is made in an app.

Reply:
I'm do agree with your concern. On Android, the onUserInteraction will also loop over the event listeners in order to fire an event. It can have performance impact on both platforms, but while testing it, I wasn't able to "feel"/measure it.

I also understand that on Android, we're firing an event for a native event (onUserInteraction). iOS doesn't have this event by default, so this workaround is mentioned as a solution for this problem (https://blog.gaelfoppolo.com/detecting-user-inactivity-in-ios-application-684b0eeeef5b)

In order to improve performance, there should be an `if` somewhere in order to only post internal notifications when there are listeners registered, but I wasn't able to find a way on how to implement this. This feature might be requested by 'only 5 devs', but I think this can't be fixed by using a module, as you need to extend the UIApplication class and register it in the main.m file of your project.

In a use case from our customer, we're in the need of implementing a way to automatically logout after x-seconds of non-activity. The way it's implemented now (firing events for user interctions) is the only way to catch user activity in Titanium, so we can handle further stuff in our Javascript code.

Reply:
If we use a design that has a property (let's say `trackUserInteraction`) together with the `userinteraction` event, we could wrap the special code out using preprocessor macros on iOS. For Android, we could maybe add the handler only if the property is set or at least use it to skip the event from firing.

Reply:
Thanks for your comment, but I don't have any knowledge about preprocessor macros. 

On Android, I think that the event is only fired when there is a listener registered.